src/snakemake/resources.py

```
@@ -536,7 +536,7 @@ def generic_callable(val, threads_arg, **kwargs):
                 args,
             )
         # Triggers for string arguments like n1-standard-4
-        except NameError:
+        except NameError or SyntaxError:
```

Ruff (0.8.2)

`except` handlers should only be exception classes or tuples of exception classes

(B030)